components/camel-mllp/src/main/java/org/apache/camel/component/mllp/MllpTcpClientProducer.java

```diff
@@ -276,6 +291,19 @@ void checkConnection() throws IOException {
                 mllpSocketWriter = new MllpSocketWriter(socket, false);
             }
         }
+        return;
+    }
+
+    @ManagedOperation(description = "Check client connection")
+    public boolean managedCheckConnection() {
+        boolean isValid = true;
+        try {
+            checkConnection();
+        } catch (IOException ioEx) {
+            isValid = false;
+            log.error("JMX check connection: {}", ioEx);
```

Comment:
well, the reason i wanted to add logging is that there might be different reasons of IOExceptions(SocketException, StreamWriting/Reading exceptions, Connection reset by peer, etc.. ) and the one who wants to manage the application might want to see the stacktrace. Agreed on that i changed to log level as debug which sounds reasonable IMHO.